L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc

```diff
@@ -688,14 +706,43 @@ void L1FPGATrackProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSe
 
   iEvent.put(std::move(L1TkTracksForOutput), "Level1TTTracks");
 
-  // produce clock and bit output tracks and stubs
-  // number of track channel
-  const int numStreamsTrack = N_SECTOR * channelAssignment_->numChannelsTrack();
-  // number of stub channel
-  const int numStreamsStub = N_SECTOR * channelAssignment_->numChannelsStub();
+  // produce clock and bit accurate stream output tracks and stubs.
+  // from end of tracklet pattern recognition.
+  // Convertion here is from stream format that allows this code to run
+  // outside CMSSW to the EDProduct one.
   Streams streamsTrack(numStreamsTrack);
   StreamsStub streamsStub(numStreamsStub);
-  eventProcessor.produce(streamsTrack, streamsStub);
+
+  for (unsigned int chanTrk = 0; chanTrk < numStreamsTrack; chanTrk++) {
+    for (unsigned int itk = 0; itk < streamsTrackRaw[chanTrk].size(); itk++) {
+      std::string bitsTrk = streamsTrackRaw[chanTrk][itk];
+      int iSeed = chanTrk % channelAssignment_->numChannelsTrack();  // seed type
+      streamsTrack[chanTrk].emplace_back(bitsTrk);
+
+      const unsigned int chanStubOffsetIn = chanTrk * maxNumProjectionLayers;
+      const unsigned int chanStubOffsetOut = channelAssignment_->offsetStub(chanTrk);
+      const unsigned int numProjLayers = channelAssignment_->numProjectionLayers(iSeed);
+      TTBV hitMap(0, numProjLayers);
+      // remove padding from stub stream
+      for (unsigned int iproj = 0; iproj < maxNumProjectionLayers; iproj++) {
+        // FW current has one (perhaps invalid) stub per layer per track.
+        const StubStreamData& stubdata = streamsStubRaw[chanStubOffsetIn + iproj][itk];
+        const L1TStub& stub = stubdata.stub();
+        if (stubdata.valid()) {
```

Comment:
I would prefer if (!valid) continue to reduce indentation

Reply:
Another matter of taste. This continue statement would be further into the loop, so more easily overlooked. Whereas the "if () {" statement with an indented block is hard to overlook.